pyrobosim/pyrobosim/navigation/astar_grid.py

```diff
@@ -0,0 +1,304 @@
+
+        path_found = False
+        timed_out = False
+        start_time = time.time()
+        # Add start node data
+        self.parent_of[self.start] = None
+        self.cost_till[self.start] = 0.0
+        expected_toal_cost = self.cost_till[self.start] + self._heuristic(
+            self.start, self.goal
+        )
+        self.candidates.put((expected_toal_cost, self.start))
+        # Search for a path
+        while not path_found and not self.candidates.empty() and not timed_out:
+            current = self._get_best_candidate()
+            if current == self.goal:
+                path_found = True
+                break
+            self._expand(current)
+            self.planning_time = time.time() - start_time
+            timed_out = self.planning_time >= self.max_time
+        # Generate the path if a path was found
+        if path_found:
+            self._generate_path(current)
+        return self.latest_path
+
+    def print_metrics(self):
+        """Print metrics about the latest path computed."""
+        if self.latest_path.num_poses == 0:
+            print("No path.")
+        else:
+            print("Latest path from A*:")
+            self.latest_path.print_details()
+        print("\n")
+        print(f"Selected actions : {self.selected_actions}")
+        print(f"Selected heuristic: {self.distance_metric}")
+        print(f"Occupancy grid generated in : {self.grid_generation_time} seconds")
+        print(f"Time to plan: {self.planning_time} seconds")
+        print(f"Number of nodes expanded : {self.num_nodes_expanded}")
+        print(f"Number of waypoints in path : {self.latest_path.num_poses}")
+
+    def plot(self, axes, path_color="m", show_path=True):
+        """
+        Plots the planned path on a specified set of axes.
+
+        :param axes: The axes on which to draw.
+        :type axes: :class:`matplotlib.axes.Axes`
+        :param path_color: Color of the path, as an RGB tuple or string.
+        :type path_color: tuple[float] / str, optional
+        :param show_graph: If True, shows the RRTs used for planning.
+        :type show_graph: bool
+        :param show_path: If True, shows the last planned path.
+        :type show_path: bool
+        :return: List of Matplotlib artists containing what was drawn,
+            used for bookkeeping.
+        :rtype: list[:class:`matplotlib.artist.Artist`]
+        """
+        artists = []
+        if self.latest_path.num_poses > 0:
+            x = [p.x for p in self.latest_path.poses]
+            y = [p.y for p in self.latest_path.poses]
+            (path,) = axes.plot(
+                x, y, linestyle="-", color=path_color, linewidth=3, alpha=0.5, zorder=1
+            )
+            (start,) = axes.plot(x[0], y[0], "go", zorder=2)
+            (goal,) = axes.plot(x[-1], y[-1], "rx", zorder=2)
+            artists.extend((path, start, goal))
+
+        return artists
+
+    def show(self, show_graph=True, show_path=True):
```

Comment:
The `show_graph` argument is not used at all, right? So we can remove it here, in the docstring, and the call to `plot()`.

Reply:
Okay

Reply:
I think the `show_path` can also be removed for both `show()` and `plot()` since that is not being used in the `plot()`.
Or is there a requirement for show to have a `show_path` argument?

Reply:
You're right, this can go too.

Based on https://github.com/sea-bass/pyrobosim/blob/main/pyrobosim/pyrobosim/gui/world_canvas.py#L316-L323, the only requirement is for the `plot()` method to accept axes as its first argument, and to have a `path_color` named arg.